File: clients/ruby/model/debug_data.rb
module Model

# TODO - Document
class DebugData

    def initialize()
    end

    # Read DebugData from input stream
    def self.read_from(stream)
        DebugData.new()
    end

    # Write DebugData to output stream
    def write_to(stream)
    end

    def to_s
        string_result = "DebugData { "
        string_result += " }"
        string_result
    end

    def to_str
        to_s
    end
end

end

File: clients/cpp/model/DebugData.cpp
#include "DebugData.hpp"

namespace model {

DebugData::DebugData() { }

// Read DebugData from input stream
DebugData DebugData::readFrom(InputStream& stream) {
    return DebugData();
}

// Write DebugData to output stream
void DebugData::writeTo(OutputStream& stream) const {
}

// Get string representation of DebugData
std::string DebugData::toString() const {
    std::stringstream ss;
    ss << "DebugData { ";
    ss << " }";
    return ss.str();
}

bool DebugData::operator ==(const DebugData& other) const {
    return true;
}

}

size_t std::hash<model::DebugData>::operator ()(const model::DebugData& value) const {
    size_t result = 0;
    return result;
}